test/expiry.test.ts

```diff
@@ -0,0 +1,83 @@
+import { store } from '../src'
+
+afterEach(() => {
+  store.clear()
+})
+
+test('does not add expired cookies', () => {
+  const req = new Request('https://mswjs.io')
+  const res = new Response(null, {
+    headers: new Headers({ 'set-cookie': `cookieName=abc-123; Expires=${ (new Date()).toUTCString() }` }),
+  })
+
+  store.add(req, res)
+
+  // Assert cookie entry has not been added.
+  const allCookies = store.getAll()
+  const allCookiesList = Array.from(allCookies.entries())
+  expect(allCookiesList).toHaveLength(0)
+
+  // Assert the cookie can't be retrieved given a matching request.
+  const reqCookies = store.get(req)
+  expect(reqCookies.size).toBe(0)
+})
+
+test('does not add cookies with a max age of zero', () => {
+  const req = new Request('https://mswjs.io')
+  const res = new Response(null, {
+    headers: new Headers({ 'set-cookie': 'cookieName=abc-123; Max-Age=0' }),
+  })
+
+  store.add(req, res)
+
+  // Assert cookie entry has not been added.
+  const allCookies = store.getAll()
+  const allCookiesList = Array.from(allCookies.entries())
+  expect(allCookiesList).toHaveLength(0)
+
+  // Assert the cookie can't be retrieved given a matching request.
+  const reqCookies = store.get(req)
+  expect(reqCookies.size).toBe(0)
+})
+
+test('does not return expired cookies', async () => {
+  const req = new Request('https://mswjs.io')
+  const date = new Date()
+  date.setMilliseconds(date.getMilliseconds() + 100)
+  const res = new Response(null, {
+    headers: new Headers({ 'set-cookie': `cookieName=abc-123; Expires=${ date.toUTCString() }` }),
+  })
+
+  store.add(req, res)
+
+  await new Promise((resolve) => setTimeout(resolve, 200))
+
+  // Assert cookie entry has been deletec.
+  const allCookies = store.getAll()
+  const allCookiesList = Array.from(allCookies.entries())
+  expect(allCookiesList).toHaveLength(0)
+
+  // Assert the cookie can't be retrieved given a matching request.
+  const reqCookies = store.get(req)
+  expect(reqCookies.size).toBe(0)
+})
+
+test('does not return cookies after the max age', async () => {
```

Comment:
Looks like we are missing a test that returns a cookie which expiration date is larger than the current date. Do you think it makes sense to add it?

Reply:
I added two more tests to cover that. I initially thought all other tests cover that scenario but they only cover the case when neither expires nor maxAge is defined.

Reply:
Thank you, @chrisguttandin!